### PR TITLE
OXT-1281: NetworkManager upgrade 1.18.4

### PR DESCRIPTION
**Pull request by mahantesh-ais**
Aligning networkmanager and network-manager-applet with zeus.

OXT-1281

### CODE REVIEW

**Comment by mahantesh-ais** [comment]
Build: [10062](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10062)

File: recipes-connectivity/networkmanager/networkmanager_1.18.4.bbappend [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    file://nm_sync.sh \
+    file://db_to_nm.awk \
+    file://nm_to_db.awk \
+    file://org.openxt.nmapplet.conf \
+    file://dbus-system.conf \
+"
+
+EXTRA_OECONF += " \
+    --disable-polkit \
+"
```
[the remark this answers is not in the document]
Comment (mahantesh-ais) [reply recipes-connectivity/networkmanager/networkmanager_1.18.4.bbappend]: Surprisingly enabling it is not breaking anything.

File: recipes-connectivity/networkmanager/networkmanager_1.18.4.bbappend [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    file://nm_sync.sh \
+    file://db_to_nm.awk \
+    file://nm_to_db.awk \
+    file://org.openxt.nmapplet.conf \
+    file://dbus-system.conf \
+"
+
+do_install_prepend () {
+    sed -i -e s:deny:allow:g ${S}/src/org.freedesktop.NetworkManager.conf
+    sed -i -e s:deny:allow:g ${S}/dispatcher/nm-dispatcher.conf
```
Comment (eric-ch) [inline recipes-connectivity/networkmanager/networkmanager_1.18.4.bbappend]: This is terrifying. From what I see it enables the following from `org.freedesktop.NetworkManager.conf`:
```xml
        <policy context="default">
```
```xml
                <deny own="org.freedesktop.NetworkManager"/>
                <deny send_destination="org.freedesktop.NetworkManager"/>
```
```xml
                <!-- Root-only functions -->
                <deny send_destination="org.freedesktop.NetworkManager"
                      send_interface="org.freedesktop.NetworkManager"
                      send_member="SetLogging"/>
                <deny send_destination="org.freedesktop.NetworkManager"
                      send_interface="org.freedesktop.NetworkManager"
                      send_member="Sleep"/>
                <deny send_destination="org.freedesktop.NetworkManager"
                      send_interface="org.freedesktop.NetworkManager.Settings"
                      send_member="LoadConnections"/>
                <deny send_destination="org.freedesktop.NetworkManager"
                      send_interface="org.freedesktop.NetworkManager.Settings"
                      send_member="ReloadConnections"/>

                <deny own="org.freedesktop.NetworkManager.dnsmasq"/>
                <deny send_destination="org.freedesktop.NetworkManager.dnsmasq"/>
```

[nm-dispatcher.conf](https://github.com/NetworkManager/NetworkManager/blob/1.18.4/dispatcher/nm-dispatcher.xml) does not have `deny` statement though.

Comment (mahantesh-ais) [reply recipes-connectivity/networkmanager/networkmanager_1.18.4.bbappend]: Yes, this is carried as legacy even in 1.4.4. But this is no longer required, I will remove it.
Thanks.

**Comment by eric-ch** [comment]
Build [$10075](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10075)

**Comment by eric-ch** [comment]
[Build $10075](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10075)

**Comment by eric-ch** [comment]
Will merge soon.

**Pull request merged**